md_test.py

```diff
@@ -0,0 +1,168 @@
+                 f"--ioengine={ioengine}",
+                 f"--numjobs={numjobs}",
+                 f"--iodepth={iodepth}",
+                 f"--offset_increment=100G",
+                 f"--output-format=json"]
+
+    # First call resets the counters
+    psutil.cpu_times_percent()
+
+    fio = run(args_list)
+
+    cpu_times = psutil.cpu_times_percent()
+
+    fio = json.loads(fio)
+    job = fio["jobs"][0]
+    rbw = job["read"]["bw"]
+    riops = job["read"]["iops"]
+    wbw = job["write"]["bw"]
+    wiops = job["write"]["iops"]
+
+    write = f"W: {suffixify(wbw)} {wiops:.0f} IOPS"
+    read = f"R: {suffixify(rbw)} {riops:.0f} IOPS"
+    cpu = f"CPU: user={cpu_times.user}% system={cpu_times.system}%"
+
+    print(f"{write} | {read} | {cpu}")
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--devices", "-d", nargs="+",
+                        help="List of devices to use. If not used, all unmounted non-Eideticom NVMe devices will be used.")
+    parser.add_argument("--verbose", "-v", action="store_true")
+    parser.add_argument("--setup-only", action="store_true",
+                        help="Only setup MD and exit (skip fio test).")
+
+    # MD options
+    md_group = parser.add_argument_group("MD options")
+    md_group.add_argument("--chunk", "-c", default="64K",
+                          help="MD chunk size")
+    md_group.add_argument("--group-thread-cnt", "-g", default=0, type=int,
+                          help="Number of threads to use for MD workqueue. Default is 0 (don't use a workqueue).")
+    md_group.add_argument("--stripe-cache-size", "-s", default=256, type=int,
+                          help="MD stripe cache size. Default is 256.")
+
+    # fio options
+    fio_group = parser.add_argument_group("fio options")
+    fio_group.add_argument("--block-size", "--bs", default="1M",
+                        help="fio block size. Default is 1M.")
+    fio_group.add_argument("--readwrite", "--rw", default="write",
+                        help="fio IO pattern. Default is \"write\".")
+    fio_group.add_argument("--numjobs", "-n", default=4, type=int,
+                        help="Number of fio jobs. Default is 4.")
+    fio_group.add_argument("--iodepth", default=2, type=int,
+                        help="fio IO depth. Default is 2.")
+    fio_group.add_argument("--runtime", "-t", default="30s",
+                        help="fio runtime. Default is 30s.")
+    args = parser.parse_args()
+
+    if os.getuid() != 0:
+        sys.exit("This script must be executed as root.")
+
+    verbose = args.verbose
+
+    devs = args.devices if args.devices else find_devices()
+
+    stop_md(devs)
+    for dev in devs:
+        clear_dev(dev)
+
+    setup_md(devs, chunk=args.chunk, group_thread_cnt=args.group_thread_cnt,
+             stripe_cache_size=args.stripe_cache_size)
+
+    if not args.setup_only:
+        run_fio(bs=args.block_size, rw=args.readwrite, numjobs=args.numjobs,
```

Comment:
Seems like we want this script to summarize and output a bunch of the metrics we care about:  read and write (separately) for sequential IO bandwidth and random IOPS. And find some optimum parameters for a fast machine.

A script that just takes in a ton of parameters for a single fio run isn't quite what we need.

Reply:
Might be overkill for this but we do have an fio python script for getting performance numbers already in: [fio.py](https://github.com/Eideticom/linux-noload/blob/master/tools/testing/selftests/noloadfs/performance/fio.py)
Don't need all the tempfile stuff when running on a block device, but it runs fio scripts passing fio arguments down in a pythonic way "fio(blocksize=...)" and returns the json formatted fio output.